runtime/service.go

```diff
@@ -1183,11 +1191,21 @@ func (s *service) shutdown(
 		s.shutdownLoop(requestCtx, timeout, req)
 	}()
 
-	err := s.machine.Wait(context.Background())
-	if err == nil {
+	var result *multierror.Error
+	if err := s.machine.Wait(context.Background()); err != nil {
+		result = multierror.Append(result, err)
+	}
+	if err := s.cleanup(); err != nil {
+		result = multierror.Append(result, err)
+	}
+	if result == nil {
```

Comment:
Do you need this? Wont the `ErrorOrNil` handle it being nil?

https://github.com/hashicorp/go-multierror/blob/master/multierror.go#L29

Reply:
That was my Java thinking... Let me fix.